File: api/src/models/models/attendanceModel.js
module.exports = (sequelize, DataTypes) => {
  const Attendance = sequelize.define('Attendance', 
  {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    tgl_mulai: {
      type: DataTypes.DATEONLY,
      allowNull: false
    },
    tgl_berakhir: {
      type: DataTypes.DATEONLY,
      allowNull: false
    },
    jlh_hari_kerja: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 30
    },
    hadir: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0
    },
    filled_at: {
      type: DataTypes.DATEONLY,
      allowNull: true
    },
    status: {
      type: DataTypes.STRING,
      allowNull: false,
      defaultValue: 'ACTIVE'
    }
  }, {
    tableName: 'attendance',
    createdAt: 'created_at',
    updatedAt: 'updated_at'
  })

  return Attendance
}

File: flutter/lib/models/attendance_model.dart
class Attendance {
  int? id;
  String? tglMulai;
  String? tglBerakhir;
  int? jlhHariKerja;
  int? hadir;
  int? idKaryawan;
  String? nama;
  String? filledAt;
  int? createdBy;
  int? updatedBy;
  bool? off;

  Attendance.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    tglMulai = json['tgl_mulai'];
    tglBerakhir = json['tgl_berakhir'];
    jlhHariKerja = json['jlh_hari_kerja'];
    hadir = json['hadir'];
    idKaryawan = json['id_karyawan'];
    nama = json['nama'];
    filledAt = json['filled_at'];
    createdBy = json['created_by'];
    updatedBy = json['updated_by'];
    off = json['off'];
  }
}